typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,440 @@
+            serviceId, 
+            containerCreationConfig.getImage(), 
+            containerCreationConfig.getUsedPortsSet(),
+            containerRunConfig.getEntrypointOverrideArgs(),
+            containerRunConfig.getCmdOverrideArgs(),
+            containerRunConfig.getEnvironmentVariableOverrides(),
+            containerCreationConfig.getKurtosisVolumeMountpoint(),
+            artifactIdStrToMountDirpath);
+        //TODO TODO TODO - CALLBACK & ERROR-HANDLING
+        // resp, err := this.client.StartService(ctx, startServiceArgs)
+        // if err != nil {
+        //     return nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
+        // }
+        log.trace("Successfully started service with Kurtosis API");
+
+        let resp: StartServiceResponse; //TODO - remove
+        return [serviceContext, resp.getUsedPortsHostPortBindingsMap(), null]
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceContext(serviceId: ServiceID): [ServiceContext, Error] {
+        const getServiceInfoArgs: GetServiceInfoArgs = newGetServiceInfoArgs(serviceId);
+        //TODO TODO TODO - CALLBACK & ERROR-HANDLING
+        // serviceResponse, err := networkCtx.client.GetServiceInfo(context.Background(), getServiceInfoArgs)
+        // if err != nil {
+        //     return nil, stacktrace.Propagate(
+        //         err,
+        //         "An error occurred when trying to get info for service '%v'",
+        //         serviceId)
+        // }
+        let serviceResponse: GetServiceInfoResponse; //TODO - Remove
+        if (serviceResponse.getIpAddr() == "") {
+            return [null, new Error(
+                "Kurtosis API reported an empty IP address for service " + serviceId +  " - this should never happen, and is a bug with Kurtosis!",
+                )
+            ]
+        }
+
+        const enclaveDataVolMountDirpathOnSvcContainer: string = serviceResponse.getEnclaveDataVolumeMountDirpath();
+        if (enclaveDataVolMountDirpathOnSvcContainer == "") {
+            return [null, new Error(
+                "Kurtosis API reported an empty enclave data volume directory path for service " + serviceId + " - this should never happen, and is a bug with Kurtosis!",
+                )
+            ]
+        }
+
+        const serviceContext: ServiceContext = ServiceContext(
+            this.client,
+            serviceId,
+            serviceResponse.getIpAddr(),
+            this.enclaveDataVolMountpoint,
+            enclaveDataVolMountDirpathOnSvcContainer,
+        )
+
+        return [serviceContext, null]
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public removeService(serviceId: ServiceID, containerStopTimeoutSeconds: number): Error {
+
+        const log: logger.Logger = new logger.Logger();
```

Comment:
Same logger comment as above. The reason the current way of creating a logger in each function isn't ideal is because: imagine we want to set the Logger to only output ERROR messages, and ignore all other messages. We'd want to configure that in only one place (likely the `main` function), but then in these functions here we'd create _new_ Loggers (which, by default, log everything of INFO level and above) so we'd have a bug where the user thinks they're only logging ERRORs but we'd be seeing INFOs here